src/mediastream.cc

```diff
@@ -0,0 +1,243 @@
+      }
+      // 4. Local MediaStream
+    }
+  }
+
+  MediaStream* mediaStream = nullptr;
+  if (!stream) {
+    auto label = rtc::CreateRandomUuid();
+    auto shouldReleaseFactory = !factory;
+    factory = factory ? factory : PeerConnectionFactory::GetOrCreateDefault();
+    stream = factory->factory()->CreateLocalMediaStream(label);
+    mediaStream = new MediaStream(std::move(factory), std::move(stream), shouldReleaseFactory);
+  } else {
+    mediaStream = new MediaStream(std::move(factory), std::move(stream));
+  }
+
+  mediaStream->Wrap(info.This());
+  mediaStream->Ref();
+
+  for (auto track : tracks) {
+    Local<Value> argv[1];
+    argv[0] = track->handle();
+    Nan::Call("addTrack", info.This(), 1, argv);
+  }
+
+  info.GetReturnValue().Set(info.This());
+}
+
+NAN_GETTER(MediaStream::GetId) {
+  auto self = Nan::ObjectWrap::Unwrap<MediaStream>(info.Holder());
+  info.GetReturnValue().Set(Nan::New(self->_stream->label()).ToLocalChecked());
+}
+
+NAN_GETTER(MediaStream::GetActive) {
+  auto self = Nan::ObjectWrap::Unwrap<MediaStream>(info.Holder());
+  auto active = false;
+  for (auto const& track : self->_stream->GetAudioTracks()) {
+    active = active ||
+        track->state() == webrtc::MediaStreamTrackInterface::TrackState::kLive;
+  }
+  for (auto const& track : self->_stream->GetVideoTracks()) {
+    active = active ||
+        track->state() == webrtc::MediaStreamTrackInterface::TrackState::kLive;
+  }
+  info.GetReturnValue().Set(Nan::New(active));
+}
+
+NAN_METHOD(MediaStream::GetAudioTracks) {
+  auto self = Nan::ObjectWrap::Unwrap<MediaStream>(info.Holder());
+  auto tracks = std::vector<MediaStreamTrack*>();
+  for (auto const& track : self->_stream->GetAudioTracks()) {
+    auto mediaStreamTrack = MediaStreamTrack::GetOrCreate(self->_factory, track);
+    tracks.push_back(mediaStreamTrack);
+  }
+  CONVERT_OR_THROW_AND_RETURN(tracks, result, Local<Value>);
+  info.GetReturnValue().Set(result);
+}
+
+NAN_METHOD(MediaStream::GetVideoTracks) {
+  auto self = Nan::ObjectWrap::Unwrap<MediaStream>(info.Holder());
+  auto tracks = std::vector<MediaStreamTrack*>();
+  for (auto const& track : self->_stream->GetVideoTracks()) {
+    auto mediaStreamTrack = MediaStreamTrack::GetOrCreate(self->_factory, track);
+    tracks.push_back(mediaStreamTrack);
+  }
+  CONVERT_OR_THROW_AND_RETURN(tracks, result, Local<Value>);
+  info.GetReturnValue().Set(result);
+}
+
+NAN_METHOD(MediaStream::GetTracks) {
+  auto self = Nan::ObjectWrap::Unwrap<MediaStream>(info.Holder());
+  auto tracks = std::vector<MediaStreamTrack*>();
+  for (auto const& track : self->_stream->GetAudioTracks()) {
+    auto mediaStreamTrack = MediaStreamTrack::GetOrCreate(self->_factory, track);
+    tracks.push_back(mediaStreamTrack);
+  }
+  for (auto const& track : self->_stream->GetVideoTracks()) {
+    auto mediaStreamTrack = MediaStreamTrack::GetOrCreate(self->_factory, track);
+    tracks.push_back(mediaStreamTrack);
+  }
```

Comment:
This is used in a couple places. Worth factoring out into a `tracks` method. Then we can reuse in `GetTracks`, `GetActive`, `GetTrackById`, etc.